query-engine/driver-adapters/src/proxy.rs

```diff
@@ -403,9 +409,23 @@ impl TransactionProxy {
     pub async fn commit(&self) -> quaint::Result<()> {
         self.commit.call(()).await
     }
+
     pub async fn rollback(&self) -> quaint::Result<()> {
         self.rollback.call(()).await
     }
+
+    pub fn discard(&self) -> quaint::Result<()> {
+        match self
+            .discard
+            .call((), napi::threadsafe_function::ThreadsafeFunctionCallMode::NonBlocking)
```

Comment:
not sure about `Blocking` vs `NonBlocking` here, Node.js docs do not describe this very well, will need to look at the sources (if we keep it at all and not replace with `AsyncJsFunction` as per comment above)